### DISCUSSION TITLE
Unresolved reference - on creating a new file

### DISCUSSION THREAD

**Discussion by JishnuGoyal** [oppia/oppia-android#4223] [Q&A (Other)] [answered]
I added a new file under the utility/profile folder called ProfileNameValidator.
I injected it in `ProfileManagementController`. It complain unresolved reference :ProfileNameValidator each time.

Detailed error:
![image](https://user-images.githubusercontent.com/64526117/155872099-729efc24-25e2-4c73-98d7-82165a4fc318.png)

Please have a look at the file from <a href="https://github.com/oppia/oppia-android/pull/4081">this</a> PR from the files changed tab.

I have created an interface and a module to make sure it gets injected properly and also written the BUILD.bazel config for this file. 
I'm not sure why this error is persisting. 
Thanks!

**Comment by BenHenning** [comment, marked as answer]
@JishnuGoyal can you add a Gist of the entire error you're experiencing, including the command line that led to it, along with the related Kotlin and Bazel files?

I suspect that this is a case of a missing dependency in the Bazel library, but I'd need to confirm.

**Reply by JishnuGoyal** [reply]
This [error](https://github.com/oppia/oppia-android/runs/5348795397?check_suite_focus=true) is from the CI result from the last push on [this](https://github.com/oppia/oppia-android/pull/4081) PR. 

The affected file is `ProfileManagmentController` due to addition of files: ProfileNameValidator, ProfileNameValidatorImpl, ProfileNameValidatorModule.

Although the app is working as expected - it receives an instance of profileNameValidator and uses it correctly, the tests are still failing. I'm not sure why bazel complains that ProfileManagmentController is not recognised. Am i introducing ProfileManagementController into the dagger graph incorrectly ?

**Reply by BenHenning** [reply]
Thanks for the context @JishnuGoyal. Confirming that this is a case of a missing dependency. https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/android/domain/profile/BUILD.bazel#L9 defines a Bazel library for ProfileManagementController. You'll need to add a dependency on the new profile_name_validator library in ProfileManagementController's ``deps`` list in order for it to be made available when building the controller.
